File: maximumSubarray/bestTimeToBuyAndSellStock2.js
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = (prices) => {
    return prices.reduce((acc, value, index) => {
        if (index < 1) return acc;
        return acc += Math.max(0, value - prices[index-1])
    }, 0);
};

File: maximumSubarray/bestTimeToBuyAndSellStock2.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        best_sum = 0 # or: float('-inf')
        for i in range(1, len(prices)):
            best_sum += max(0, prices[i] - prices[i-1])
        return best_sum
        